sdk/src/contexts/solana/utils/tokenBridge/instructions/redeemAndUnwrap.ts

```diff
@@ -0,0 +1,107 @@
+  Keypair,
+  SystemProgram,
+  Transaction,
+} from '@solana/web3.js';
+import { createCompleteTransferNativeInstruction } from '@certusone/wormhole-sdk/lib/esm/solana/tokenBridge';
+import { addComputeBudget } from '../../computeBudget';
+
+// This is taken from SDKv1 and removes the `partialSign` call
+// so Connect can set a compute budget before signing
+export async function redeemAndUnwrapOnSolana(
+  connection: Connection,
+  bridgeAddress: PublicKeyInitData,
+  tokenBridgeAddress: PublicKeyInitData,
+  payerAddress: PublicKeyInitData,
+  signedVaa: SignedVaa,
+  commitment?: Commitment,
+) {
+  const parsed = parseTokenTransferVaa(signedVaa);
+  const targetPublicKey = new PublicKey(parsed.to);
+  const targetAmount = await getMint(connection, NATIVE_MINT, commitment).then(
+    (info) =>
+      parsed.amount * BigInt(Math.pow(10, info.decimals - MAX_VAA_DECIMALS)),
+  );
+  const rentBalance = await getMinimumBalanceForRentExemptAccount(
+    connection,
+    commitment,
+  );
+  if (Buffer.compare(parsed.tokenAddress, NATIVE_MINT.toBuffer()) != 0) {
+    return Promise.reject('tokenAddress != NATIVE_MINT');
+  }
+  const payerPublicKey = new PublicKey(payerAddress);
+  const ancillaryKeypair = Keypair.generate();
+
+  const completeTransferIx = createCompleteTransferNativeInstruction(
+    tokenBridgeAddress,
+    bridgeAddress,
+    payerPublicKey,
+    signedVaa,
+  );
+
+  //This will create a temporary account where the wSOL will be moved
+  const createAncillaryAccountIx = SystemProgram.createAccount({
+    fromPubkey: payerPublicKey,
+    newAccountPubkey: ancillaryKeypair.publicKey,
+    lamports: rentBalance, //spl token accounts need rent exemption
+    space: ACCOUNT_SIZE,
+    programId: TOKEN_PROGRAM_ID,
+  });
+
+  //Initialize the account as a WSOL account, with the original payerAddress as owner
+  const initAccountIx = createInitializeAccountInstruction(
+    ancillaryKeypair.publicKey,
+    NATIVE_MINT,
+    payerPublicKey,
+  );
+
+  //Send in the amount of wSOL which we want converted to SOL
+  const balanceTransferIx = createTransferInstruction(
+    targetPublicKey,
+    ancillaryKeypair.publicKey,
+    payerPublicKey,
+    targetAmount.valueOf(),
+  );
+
+  //Close the ancillary account for cleanup. Payer address receives any remaining funds
+  const closeAccountIx = createCloseAccountInstruction(
+    ancillaryKeypair.publicKey, //account to close
+    payerPublicKey, //Remaining funds destination
+    payerPublicKey, //authority
+  );
+
+  const { blockhash } = await connection.getLatestBlockhash(commitment);
+  const transaction = new Transaction();
+  transaction.recentBlockhash = blockhash;
+  transaction.feePayer = payerPublicKey;
+  transaction.add(
+    completeTransferIx,
+    createAncillaryAccountIx,
+    initAccountIx,
+    balanceTransferIx,
+    closeAccountIx,
+  );
+  // Set the compute budget before signing
+  await addComputeBudget(connection, transaction, undefined, 0.9, 500_000);
```

Comment:
We set a floor of 500k microlamports for the last transaction when redeeming WSOL on Solana.